src/osd/PGBackend.cc

```diff
@@ -866,8 +866,6 @@ map<pg_shard_t, ScrubMap *>::const_iterator
       auth = j;
       *auth_oi = oi;
       auth_version = oi.version;
-      auth_bl.clear();
```

Comment:
@dzafman but David, the existing logic reset `auth_bl` with the current `bl`, as long as it's "better" by the criteria above. while this PR always stick with the first one. i know, it's more consistent this way, because instead of using the "best" bl so far which might change when we find a better one, we are always comparing the bl with the *first* non-empty OI attr bl. but again, the first non-empty is not necessarily the best OI attr bl.